cats/src/main/scala/scalaz/ioeffect/catz.scala

```diff
@@ -6,24 +6,25 @@ import cats.syntax.all._
 
 import scala.util.control.NonFatal
 
-object catz extends RTS {
+object catz extends RTS with IOEffectInstance
+private[ioeffect] trait IOEffectInstance {
```

Comment:
Why private? What about if $USER wants to reuse the instance but mixin a customized `RTS`?

What is the criteria for picking `trait` vs `abstract class`?

Reply:
Then downstream needs to put their code in this package... it's internal, not part of the public API, so applications can use it but not libraries.

Reply:
always pick abstract class when you can. Yeah I should have used abstract class. It's easy to forget.

Reply:
Fuck inheritance. It is sad but I think with an implicit `RTS` it would be more "elegant". Sorry :grimacing:

Reply:
I'd be ok with an explicit RTS, but this is a mechanism to allow people to write their own RTS...